packages/core/src/command/hl7v2-subscriptions/hl7v2-roster-generator.ts

```
@@ -102,30 +111,124 @@ export class Hl7v2RosterGenerator {
     const rosterCsv = this.generateCsv(rosterRows);
     log("Created CSV");
 
-    const fileName = createFileKeyHl7v2Roster(hieName);
-
-    await storeInS3WithRetries({
-      s3Utils: this.s3Utils,
-      payload: rosterCsv,
-      bucketName: this.bucketName,
-      fileName,
-      contentType: CSV_MIME_TYPE,
-      log,
-      errorConfig: {
-        errorMessage: "Error uploading patient roster CSV",
-        context: "Hl7v2RosterGenerator",
-        captureParams: loggingDetails,
-        shouldCapture: true,
-      },
-    });
+    const fileName = createFileNameHl7v2Roster(hieName);
+    let failedStage: FailedStage = undefined;
+
+    try {
```

try catch-ing code everywhere is a design smell. It's more common to throw and let the error bubble up to a single try catch wrapper that knows how to handle the error. In this case, I'd define a private method, maybe _execute() and then define the public execute as
```
  async execute(config: HieConfig | VpnlessHieConfig): Promise<string> {
     try { 
       await this._execute(config)
     } catch (error) {
       if (/*error is of MyErrorType*/) { 
         const errorStage = error.stage
         ourUsualErrorLogger(error)
         this.logResults(/* stuff */)
       } else {
         /* boring normal error handling code */
       }
     }
}
```

---

Agreed. I think it is unnecessary now. Removed try catch to let errors bubble